### PR TITLE
Make TensorPipe the default backend for RPC

### PR DESCRIPTION
**Pull request by mrshenli**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43337 Minor RPC doc fixes
* **#43246 Make TensorPipe the default backend for RPC**

Differential Revision: [D23206042](https://our.internmc.facebook.com/intern/diff/D23206042)

### CODE REVIEW

File: docs/source/rpc.rst [hunk 1]
```diff
@@ -231,6 +189,48 @@ Example::
     :inherited-members:
 
 
+Process Group Backend
+"""""""""""""""""""""
+
+The Process Group agent instantiates a process group from
+the :mod:`~torch.distributed` module and utilizes its point-to-point
+communication capabilities to send RPC messages across. Internally, the process
```
Comment (osalpekar) [inline docs/source/rpc.rst:197]: nit: we can probably remove the word "across"

**Review by osalpekar — APPROVED** [review]
LGTM

File: docs/source/rpc.rst (hunk 1 above)
Comment (mrshenli) [reply docs/source/rpc.rst:197]: edited in the PR on top.

**Review by jiayisuse — APPROVED** [review]
Great to move RPC to tensorpipe! Let's wait for CI signals :)

**Pull request closed without merging**